File: Javascript/Algorithms/965.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
let isUnivalTree = root => {
    const stack = [root],
        value = root.val;
    
    while (stack.length){
        let node = stack.pop();
        let left = node.left,
            right = node.right;
        
        if (right){
            if (right.val != value){
                return false;
            }
            stack.push(right);
        }
        if (left){
            if (left.val != value){
                return false;
            }
            stack.push(left);
        }
    }
    return true;
};

File: Python/Algorithms/965.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isUnivalTree(self, root) -> bool:       
        value, stack = root.val, [root]
        push, pop = stack.append, stack.pop
        while stack:
            node = pop()
            # if value is the same, push on stack,
            # else, bail.
            for child in (node.left, node.right):
                if child:
                    if child.val != value:
                        return False
                    push(child)
        return True
